modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GmosSController.scala

```diff
@@ -0,0 +1,64 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.seqexec.model.dhs.ImageFileId
+import squants.Length
+import scala.concurrent.duration.Duration
+
+trait GmosSController {
+  import GmosSController._
+
+  // I'm not sure if getConfig will be used. It made sense for TCS, because parts of the TCS configuration cannot be
+  // inferred from the sequence, and because Seqexec needs to temporarily change parts of the TCS configuration only to
+  // later revert those changes to the previous values. But for most (if not all) instruments, the sequence completely
+  // defines the instrument configuration.
+  def getConfig: SeqAction[GmosSConfig]
+
+  def applyConfig(config: GmosSConfig): SeqAction[Unit]
+
+  def observe(obsid: ImageFileId): SeqAction[ImageFileId]
+}
+
+object GmosSController {
+
+  type Filter = edu.gemini.spModel.gemini.gmos.GmosSouthType.FilterSouth
+  type FPUSouth = edu.gemini.spModel.gemini.gmos.GmosSouthType.FPUnitSouth
```

Comment:
I don't think you need to put "South" on all these names. The fact that they are defined inside GmosSController already gives them a separate namespace, clearly identifiable with GMOS South.